build/generate-style-spec.ts

```diff
@@ -122,6 +124,37 @@ fs.writeFileSync('src/types.g.ts',
 
 export type ColorSpecification = string;
 
+export type PrimitiveProjection = 'mercator' | 'globe';
+export type ProjectionTransition = [PrimitiveProjection, PrimitiveProjection, number];
+export type ProjectionType = PrimitiveProjection | ProjectionTransition | PropertyValueSpecification<ProjectionTransition>
+export type ProjectionSpecification = {"mode":ProjectionType}
+
+
+
+export function isProjectionType(value: unknown): value is ProjectionType {
```

Comment:
This file generates code, I don't think it should have methods here, these methods can be written in regular typescript files that can hold this logic.